cardano/src/block/block.rs

```diff
@@ -94,6 +95,13 @@ impl chain_core::property::Header for BlockHeader {
             BlockHeader::MainBlockHeader(ref header) => header.consensus.slot_id.into(),
         }
     }
+
+    fn version(&self) -> Self::Version {
+        match self {
+            BlockHeader::BoundaryBlockHeader(ref header) => unimplemented!(),
```

Comment:
This will crash on any boundary block?

Reply:
Think of returning `Option<Self::Version>` if Cardano boundary blocks can't have version by design.